Dockerfile

```diff
@@ -0,0 +1,57 @@
+ARG TOMCAT_VERSION=9.0.38
+ARG JAVA_VERSION=8
+ARG IMAGE_VERSION=${TOMCAT_VERSION}-jdk${JAVA_VERSION}-openjdk-slim
+
+FROM tomcat:$IMAGE_VERSION
+
+ARG GS_VERSION=2.18.0
+ARG MARLIN_TAG=0_9_4_3
+ARG MARLIN_VERSION=0.9.4.3
+
+ARG WAR_URL=https://downloads.sourceforge.net/project/geoserver/GeoServer/${GS_VERSION}/geoserver-${GS_VERSION}-war.zip
+ARG STABLE_PLUGIN_URL=https://sourceforge.net/projects/geoserver/files/GeoServer/${GS_VERSION}/extensions
+
+# environment variables
+ENV GS_VERSION=${GS_VERSION} \
+    MARLIN_TAG=${MARLIN_TAG} \
+    MARLIN_VERSION=${MARLIN_VERSION} \
+    WAR_URL=${WAR_URL} \
+    STABLE_PLUGIN_URL=${STABLE_PLUGIN_URL} \
+    INITIAL_MEMORY="2G" \
+    MAXIMUM_MEMORY="4G" \
+    JAIEXT_ENABLED=true \
+    STABLE_EXTENSIONS='' \
+#    COMMUNITY_EXTENSIONS='' \
+    DEBIAN_FRONTEND=noninteractive \
+    EXTENSION_DOWNLOAD_DIR=/opt/geoserver_extension_downloads \
+    GEOSERVER_DATA_DIR=/opt/geoserver_data \
+    GEOWEBCACHE_CACHE_DIR=/opt/geowebcache_data
```

Comment:
you may want to expose these as a volume so the can be backed up and migrated

Reply:
The GEOSERVER_DATA_DIR is handy for outside access, easy way to stage data for publication.

Reply:
Yes. These directories should be mounted. I'll soon add a docker-compose demo for this